Unable to build project referencing latest .NET 9 Preview 6 Microsoft.NET.ILLink package

We believe the latest .NET 9 preview versions of Microsoft.NET.ILLink may contain a packaging error. This nuget now declares a package dependency on `ILCompiler.DependencyAnalysisFramework`, which doesn't appear to be available:

> D:\a\_work\1\s\src\Microsoft.Android.Sdk.ILLink\Microsoft.Android.Sdk.ILLink.csproj : error NU1101: Unable to find package ILCompiler.DependencyAnalysisFramework. No packages exist with this id in source(s): darc-pub-dotnet-emsdk-fcd8602, darc-pub-dotnet-runtime-0ece505, dotnet-eng, dotnet-public, dotnet-tools, dotnet8, dotnet8-transport, dotnet9, dotnet9-transport, xamarin.android util

--- a/src/tools/illink/src/linker/Mono.Linker.csproj
+++ b/src/tools/illink/src/linker/Mono.Linker.csproj
@@ -77,7 +77,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="$(CoreClrProjectRoot)tools\aot\ILCompiler.DependencyAnalysisFramework\ILCompiler.DependencyAnalysisFramework.csproj" />
+    <!-- PrivateAssets prevents ILCompiler.DependencyAnalysisFramework from becoming a dependency of the Microsoft.NET.ILLink package, which is designed to
+         be referenced by custom step authors as the public surface available to custom steps (which should not use the dependency analysis framework). -->
+    <ProjectReference Include="$(CoreClrProjectRoot)tools\aot\ILCompiler.DependencyAnalysisFramework\ILCompiler.DependencyAnalysisFramework.csproj" PrivateAssets="All" />
   </ItemGroup>
 
   <ItemGroup>

